aip/0008.md

```diff
@@ -174,6 +174,198 @@ When AIPs reference other AIPs, the prosaic text **must** use the format
+(Methods: GET) to reference AIP-121 (Resource Names), as the address used by
+the method should follow the guidelines for resource naming. It may also be
+possible for each of these AIPs to be overridden at an organizational level,
+yet still maintain their relationship one with another.
+
+In such a case, it would be important that the links between these two AIPs
+took into consideration that context under which they were being viewed. For
+example if AIP-131 was being viewed under the organization context, then all
+links to AIP-121 should point to the organizational extension of AIP-121, and
+not the domain level implementation of AIP-121. Likewise, if AIP-121 is being
+viewed at the domain level, then links to AIP-131 should point to the domain
+level implementation of AIP-131, and not to any organizational extensions.
+
+All AIP reference links that need to take context into consideration **must**
+be implemented as follows:
+
+Note: Still brainstorming some ideas, leave feedback:
+
+#### Includes
+
+Idea here is to use a custom \_includes file that will generate the url based
+on whether or not there is a context level implementation or not.
+
+**What the AIP Editor Adds**
+
+```
+{{ include aip_ref.html aip=131}}
+```
+
+**What the solution looks like** This is fairly pseudo codey. I confirmed
+there's a way to do this, but am not going to go write perfect code
+
+```
+<script>
+  const context = localStorage.get("context")
+  if (context and {{ file_exists("/"$context"/"aip)}})
+    window.getElementById("{{ aip }}").setAttribute("href", `/${context}/aip.md`)
+  else
+    window.getElementById("{{ aip }}").setAttribute("href", `./aip.md`)
+</script>
+<a id="{{ aip }}">AIP-{{ aip }}</a>
+```
+
+Pros:
+
+- Pretty clean implementation from the client's perspective
+- Implementing a static reference would be straight forward
+
+Cons
+
+- utilizes includes, which is slow
+
+#### Automagic
+
+Automatically looks at local refs to see if they exist.
+
+**What the AIP Editor Adds**
+
+```
+./0131.md
+```
+
+**What the solution looks like** Javascript file that is added to each aip
+file:
+
+```
+$.when($.ready)(() => {
+  const context = localStorage.get("context")
+
+  const anchors = window.getElementById("aip-main").getElementsByTagName("a")
+  let href
+  for(i = 0; i anchors.length; i++) {
+    href = anchors.getAttribute("href")
+
+    if( href is relative) {
+      if({{ file_exits("/"context"/"href) }})
+        use /context/href
+      else
+        use ./href
+    }
+  }
+})
+```
+
+Pros:
+
+- client does not have to do anything for context based references
+
+Cons:
+
+- client will need to do something different for static references
+- javascript iterates over all the anchors in the aip-main element, and
+  potentially alters them. This is just more overhead when loading a page. Not
+  huge, but still there.
+- feels a little hackish.
+
+#### Class Based
+
+Takes advantage of a kramdown feature for providing css classes to links
```

Comment:
```suggestion
Takes advantage of a [kramdown](https://kramdown.gettalong.org/) feature for providing css classes to links
```
Without a link, it can be considered a typo.